### PR TITLE
Allow update sites to define a custom URL for wizard plugin suggestion

### PR DESCRIPTION
**Pull request by olamy**
Use case: a unique distribution with two json files: 
- the usual one,
- a second one with a limited list. Some plugins maybe not be compatible with the startup mode (e.g FIPS mode)

currently the file is hardcoded and there is not extension point available with the setup wizard

### Proposed changelog entries

- Add ability for custom update center's to override the suggested plugin list

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by timja — COMMENTED** [review]
Can you provide a PR to jenkins.io with the system property please?

https://github.com/jenkins-infra/jenkins.io/edit/master/content/doc/book/managing/system-properties.adoc

**Comment by olamy** [comment]
@timja https://github.com/jenkins-infra/jenkins.io/pull/7068

**Review by timja — APPROVED** [review]

File: core/src/main/java/jenkins/install/SetupWizard.java
Comment (jtnord) [inline core/src/main/java/jenkins/install/SetupWizard.java:564]: given this needs co-operation from the actual update site to provide the file, would it not be better to obtain the URL from the `UpdateSite`  (introduce a new method that does the same as today and then anyone that needs to override it can do so)  (after all if the Jenkins project decides to offer a different list of recommended plugins in FIPS mode, I would expect that to be driven by the FIPS flag and the UpdateSite make a switch, and a user would not need to provide 2 switches).

This also only works for a hosted (http(s)) provider and the [fallback does not use the property](https://github.com/jenkinsci/jenkins/blob/569c8b0f7fb5327a7d3ea832359c8ad7aa2ea9b8/core/src/main/java/jenkins/install/SetupWizard.java#L621-L631)

Comment (olamy) [reply core/src/main/java/jenkins/install/SetupWizard.java]: yup sounds better to have only one flag to use and avoid adding too many system properties.
I changed to add a method to UpdateSite

File: core/src/main/java/hudson/model/UpdateSite.java
Comment (olamy) [inline core/src/main/java/hudson/model/UpdateSite.java:499]: because I'm not sure when this will be merged.. :)

Comment (timja) [reply core/src/main/java/hudson/model/UpdateSite.java:499]:
```suggestion
     * @since TODO
```

**Review by daniel-beck — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by timja** [comment]
I've reworded the changelog can you please take a look @olamy

**Review by jtnord — APPROVED** [review]

**Comment by olamy** [comment]
> I've reworded the changelog can you please take a look @olamy

Thanks

**Pull request merged**
